Raven.Database/Raft/ClusterManager.cs

```diff
@@ -35,14 +37,70 @@ public class ClusterManager : IDisposable
 
         public bool HasNonEmptyTopology => Engine.CurrentTopology.AllNodes.Any();
 
-        private Timer timer = null;
+        private Timer timer;
 
         public ClusterManager(RaftEngine engine, DatabasesLandlord databasesLandlord)
         {
             Engine = engine;
             Client = new ClusterManagementHttpClient(engine);
             DatabasesLandlord = databasesLandlord;
             engine.StateChanged += OnRaftEngineStateChanged;
+            engine.ProposingCandidacy += OnProposingCandidacy;
+            maxReplicationLatency = DatabasesLandlord.SystemDatabase.Configuration.Cluster.MaxReplicationLatency;
+            clusterManagerStartTime = DateTime.Now;
+        }
+
+        private readonly TimeSpan maxReplicationLatency;
+        private readonly DateTime clusterManagerStartTime;
+        private void OnProposingCandidacy(object sender, ProposingCandidacyResult e)
+        {
+            var replicationStateDoc = DatabasesLandlord.SystemDatabase.Documents.Get(Constants.Cluster.ClusterReplicationStateDocumentKey,null);
+            if (replicationStateDoc == null)
+            {
+                //This is a case of a node loading for the first time and just never got any replication state.
+                //If we prevent this than a cluster will be non-respnosive when loaded (mostly a test senario but could be a real issue)
+                if (clusterManagerStartTime + maxReplicationLatency + maxReplicationLatency < DateTime.Now)
+                {
+                    e.VetoCandidacy = true;
+                    e.Reason = "Could not find replication state document";                    
+                }
+                return;
+            }
+            var replicationState = replicationStateDoc.DataAsJson.ToObject<ReplicationState>();
```

Comment:
What meaningful thing will you do here?
Also, I've some fixes already, don't do work until I push, please.